extras/nb_table_exporter.py

```diff
@@ -119,5 +99,111 @@ def get_instruction_id(t_id, connection):
+
+
+def export_all_device_records():
+    conn = get_conn("neurobooth", False)
+    path = os.path.join(write_path, 'devices')
+
+    task_ids = get_task_ids_for_collection("test_mvp_030", conn)
+    for t_id in task_ids:
+        table_task = Table("nb_task", conn=conn)
+        task_df = table_task.query(where=f"task_id = '{t_id}'")
+        (device_ids,) = task_df["device_id_array"]
+
+        table_device = Table("nb_device", conn=conn)
+        for device_id in device_ids:
+            device_df = table_device.query(where=f"device_id = '{device_id}'")
+            (device_sn,) = device_df["device_sn"]
+            (device_name,) = device_df["device_name"]
+            (device_location,) = device_df["device_location"]
+            (wearable,) = device_df["wearable_bool"]
+            (device_make,) = device_df["device_make"]
+            (device_model,) = device_df["device_model"]
+            (device_firmware,) = device_df["device_firmware"]
+            (sensor_id_array,) = device_df["sensor_id_array"]
+
+            dev_dict = {}
+            dev_dict['device_id'] = device_id
+            dev_dict['device_sn'] = device_sn
+            dev_dict['device_name'] = device_name
+            dev_dict['device_location'] = device_location
+            dev_dict['wearable_bool'] = wearable
+            dev_dict["device_make"] = device_make
+            dev_dict["device_model"] = device_model
+            dev_dict["device_firmware"] = device_firmware
+            dev_dict["sensor_ids"] = sensor_id_array
+            dev_dict['arg_parser'] = 'iout.stim_param_reader.py::DeviceArgs'
+
+            filename = os.path.join(path, device_id + ".yml")
+            with open(filename, 'w') as f:
+                yaml.dump(dev_dict, f, sort_keys=False)
+
+
+def export_all_sensor_records():
+    conn = get_conn("neurobooth", False)
+    path = os.path.join(write_path, 'sensors')
+
+    table_sens = Table("nb_sensor", conn=conn)
+    sens_df = table_sens.query()
+    sens_df.reset_index()
+    for index, row in sens_df.iterrows():
+        sensor_id = index
+        temporal_res = row["temporal_res"]
+        spatial_res_x = row["spatial_res_x"]
+        spatial_res_y = row["spatial_res_y"]
+        file_type = row["file_type"]
+        additional_parameters = row["additional_parameters"]
+
+        sens_dict = {}
+        sens_dict["sensor_id"] = sensor_id
+        if not math.isnan(temporal_res):
+            sens_dict["temporal_res"] = temporal_res
+        if not math.isnan(spatial_res_x):
+            sens_dict["spatial_res_x"] = spatial_res_x
+        if not math.isnan(spatial_res_y):
+            sens_dict["spatial_res_y"] = spatial_res_y
+        sens_dict["file_type"] = file_type
+        sens_dict['arg_parser'] = 'iout.stim_param_reader.py::SensorArgs'
+
+        if additional_parameters is not None:
+            for key in additional_parameters:
+                sens_dict[key] = additional_parameters[key]
+
+        print(sens_dict)
```

Comment:
Intended, or leftover debug?

Reply:
I will remove this module when this work is done since all these tables will be gone